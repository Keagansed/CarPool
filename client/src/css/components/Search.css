@import url(//netdna.bootstrapcdn.com/font-awesome/4.0.3/css/font-awesome.css);

body {
    padding: 30px 0px 60px;
}

.panel > .list-group .list-group-item:first-child {
    border-top: 1px solid rgb(204, 204, 204);
}

@media (max-width: 767px) {
    .visible-md {
        display: inline-block !important;
    }
    .block {
        display: block !important;
        width: 100%;
        height: 1px !important;
    }
}
#back-to-bootsnipp {
    position: fixed;
    top: 10px; right: 10px;
}


.c-search > .form-control {
   border-radius: 0px;
   border-width: 0px;
   border-bottom-width: 1px;
   font-size: 1.3em;
   padding: 12px 12px;
   height: 44px;
   outline: none !important;
}
.c-search > .form-control:focus {
    outline:0px !important;
    -webkit-appearance:none;
    box-shadow: none;
}

.c-search > .input-group-btn .btn {
   border-radius: 0px;
   border-width: 0px;
   border-left-width: 1px;
   border-bottom-width: 1px;
   height: 44px;
}

.c-list {
    padding: 0px;
    min-height: 44px;
}
.title {
    display: inline-block;
    font-size: 1.7em;
    font-weight: bold;
    padding: 5px 15px;
}

img{
  width: 75px;
  height: 75px;
}

li{
  border: 1px solid lightgray;
  background-color: #eff2f2;
  font-size: 18px;
  height: 76px;
}

.userName{
  padding-left: 10px;
}

ul{
  list-style-type: none;
  padding: 5px 15px;
}

ul.c-controls {
    list-style: none;
    margin: 0px;
    min-height: 44px;
}

ul.c-controls li {
    margin-top: 8px;
    float: left;
}

ul.c-controls li a {
    font-size: 1.7em;
    padding: 11px 10px 6px;   
}
ul.c-controls li a i {
    min-width: 24px;
    text-align: center;
}

ul.c-controls li a:hover {
    background-color: rgba(51, 51, 51, 0.2);
}

.c-toggle {
    font-size: 1.7em;
}

.name {
    font-size: 1.7em;
    font-weight: 700;
}

.c-info {
    padding: 5px 10px;
    font-size: 1.25em;
}