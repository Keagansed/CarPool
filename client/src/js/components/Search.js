import React, { Component } from 'react';
import 'bootstrap/dist/css/bootstrap.min.css'; 

import "../utils/searchQuery.js";
import '../../css/App.css';

class Search extends Component {

	render(){
		
		return(

            <div className="center_container search_center_container">
				<br />
				<div className="bubble">
                    <div className="row no-padding-margin">
			            <div className="col-md-8 offset-md-2">
			            	<form id="searchUsers">
				                <div className="row">
				                    <div className="col-md-8 col-8 no-padding-margin">
				                        <div className="input-group c-search">
				                            <input type="text" className="form-control" id="user-search"/>
				                        </div>
				                    </div>
				                    <div className="col-md-4 col-4 no-padding-margin">
				                    	<button type="submit" className="btn btn-info" >Search</button>								
				                    </div>
				                </div> <br />
				            </form>
                        </div>
                    </div>
                    <div className="row">
                        <div id="user-list" className="col-md-8 offset-md-2">			{/*results get added to this list*/}
                            {/*<li className="list-group-item">
			                        <div className="col-md-8">
			                           <span className="name"></span><br/>
			                        </div>
			                    </li>*/}
                        </div>
                    </div>
				</div>
			</div>
		);
	}
}

export default Search;