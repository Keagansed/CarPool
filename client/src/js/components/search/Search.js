// File Type: Component

import React, { Component } from 'react';

//import { searchUsers } from "../utils/searchQuery.js";
import SearchResultList from './SearchResultList.js'

/**
* Purpose: Search input/output component. Will search for username and list all results
*/
class Search extends Component {
	constructor() {
		super();

		this.state = {
			searchResults:""
		};
	}

	handleChange = (event) => {
		// searchQuery is missing... might have been deleted during file clean up

		// searchUsers(event.target.value).then(res => {
		// 	this.setState({searchResults : res});
		// });
	}
	
	render() {		
		return(			
            <div className="center_container search_center_container">
				<div className="row no-padding-margin search-container">
					<div className="col-md-8 offset-md-2 no-padding">
						<form id="searchUsers">
							<div className="row">
								<div className="col-md-12 col-12 no-padding-margin">
									<div className="input-group c-search">
										<input 	
											type="text" 
											className="form-control" 
											id="user-search" 
											onChange={this.handleChange} 
											autoComplete="off" 
											placeholder="Search"
										/>
									</div>
								</div>
							</div>
						</form>
						<SearchResultList results={this.state.searchResults}/> {/* Search results injected into this component*/}	                		                
					</div>
				</div>
			</div>
		);
	}
}

export default Search;
