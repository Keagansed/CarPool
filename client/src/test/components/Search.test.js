import React from 'react';
import Search from '../../js/components/Search';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('Search Component', () => {
    let container;

    beforeEach(() => {
      container = shallow(<Search />);
    });

    it('captures snapshot', () => {
      const renderedValue = renderer.create(<Search />).toJSON();
      expect(renderedValue).toMatchSnapshot();
    });
    
    it ('renders correctly', () => {
      expect(container.length).toEqual(1);
    });
});
