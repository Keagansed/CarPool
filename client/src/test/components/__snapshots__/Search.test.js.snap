// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component captures snapshot 1`] = `
<div
  className="center_container search_center_container"
>
  <div
    className="row no-padding-margin search-container"
  >
    <div
      className="col-md-8 offset-md-2 no-padding"
    >
      <form
        id="searchUsers"
      >
        <div
          className="row"
        >
          <div
            className="col-md-12 col-12 no-padding-margin"
          >
            <div
              className="input-group c-search"
            >
              <input
                autoComplete="off"
                className="form-control"
                id="user-search"
                onChange={[Function]}
                placeholder="Search"
                type="text"
              />
            </div>
          </div>
        </div>
      </form>
      <div />
       
    </div>
  </div>
</div>
`;
